### PR TITLE
Implement `cudf::group_by` (sort) for `decimal32` and `decimal64`

### PR DESCRIPTION
**Pull request by codereport**
This PR resolves a part of https://github.com/rapidsai/cudf/issues/3556.

I decided to push the changes for sort `cudf::group_by` and hash `group_by` in different PRs.

### CODE REVIEW

File: cpp/src/aggregation/aggregation.cu [hunk 1]
```diff
@@ -27,8 +27,13 @@ void initialize_with_identity(mutable_table_view& table,
...
   // kernel per column
   for (size_type i = 0; i < table.num_columns(); ++i) {
-    auto col = table.column(i);
-    dispatch_type_and_aggregation(col.type(), aggs[i], identity_initializer{}, col, stream);
+    auto col        = table.column(i);
+    auto const type = [&] {
+      if (not is_fixed_point(col.type())) return col.type();
+      return col.type().id() == type_id::DECIMAL32 ? data_type{type_id::INT32}
+                                                   : data_type{type_id::INT64};
+    }();
```
Comment (jrhemstad) [inline cpp/src/aggregation/aggregation.cu:35]: Can we just have a runtime `type_id` equivalent of `device_storage_type`?

Comment (codereport) [reply cpp/src/aggregation/aggregation.cu:35]: Yea, I had thought about that. Wasn't sure it was worth it as this _currently_ is the only place this is used. Happy to add one though.

**Review by rgsl888prabhu — APPROVED** [review]
LGTM, just have to update copyright.

File: cpp/rmm_log.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+[ 23294][23:00:25:246785][info  ] ----- RMM LOG BEGIN [PTDS DISABLED] -----
```
Comment (karthikeyann) [inline cpp/rmm_log.txt:1]: What is this file?

File: cpp/include/cudf/detail/aggregation/aggregation.cuh [hunk 3]
```diff
@@ -432,6 +432,8 @@ struct identity_initializer {
   template <typename T, aggregation::Kind k>
   static constexpr bool is_supported()
   {
+    // Note: !is_fixed_point<T>() means that aggregatations for fixed_point should happen on the
```
Comment (karthikeyann) [inline cpp/include/cudf/detail/aggregation/aggregation.cuh:435]:
```suggestion
    // Note: !is_fixed_point<T>() means that aggregations for fixed_point should happen on the
```

**Review by karthikeyann — CHANGES_REQUESTED** [review]

File: cpp/rmm_log.txt (hunk 2 above)
Comment (harrism) [reply cpp/rmm_log.txt:1]: Generated by RMM, but only in the current directory of execution... We should add it to .gitignore

**Review by karthikeyann — APPROVED** [review]

**Comment by codereport** [comment]
@gpucibot merge

**Comment by codereport** [comment]
@gpucibot merge

**Pull request merged**